### ISSUE TITLE
FeignClient w/ hystrix/ribbon, GET param is double URL encoded

### ISSUE THREAD

**Issue by bitsofinfo** [spring-cloud/spring-cloud-netflix#1245]
I have a @FeignClient that is passing a GET parameter w/ a @ symbol in it, the endpoint is reporting that it received the @ as a %2540 (which means something is double escaping this). The character is a @ before it enters the feign client, but the response from the called service comes back with an error that my data could not be found because what it received was %2540

When debugging, my values is `some@emailaddr.com` as it enters the @FeignClient instance. It then gets pre encoded by `feign.RequestTemplate.replaceQueryValues()`. (i.e. `...&email=some%40emailaddr.com`). 

Once however it enters this method at: 

`org.springframework.cloud.netflix.feign.ribbon.FeignLoadBalancer.reconstructURIWithServer()` it 

Hits this line of code, which re-encodes the value so I now have `...&email=some%2540emailaddr.com` in the final url that is being executed

`UriComponentsBuilder.fromUri(original).scheme("https").build().toUri();` 

my interface declaration looks like this: 

```
@FeignClient(name="my-endpoint", fallback = UserClientFallback.class)
public interface UserClient {

	@RequestMapping(method = RequestMethod.GET, value = "/users/search/findByEmail?projection=minimal", consumes = "application/json")
    public Object findByEmail(@RequestParam("email") String email);
}
```

Maybe related? https://github.com/spring-cloud/spring-cloud-netflix/issues/1241 

Looks very similar. In ServerIntrospector, my PortType gleamed via discovery of the endpoint it is trying to reach == SECURE... so I don't think the workaround presented at #1241 (i.e. custom ServerIntrospector bean will work for me)

**Comment by bitsofinfo** [comment]
Any thoughts on this one?

**Comment by spencergibb** [comment]
There was a recent fix where changing `.build()` to `.build(true)` was the fix. How about a PR with your test?

**Comment by spencergibb** [comment]
Closed via 43adc84dfdfc33f3be0268e09b193afc82f9bf9a
